dss/operations/checkout.py

```diff
@@ -0,0 +1,191 @@
+from cloud_blobstore import BlobNotFoundError
+from dss.api.bundles import get_bundle_manifest
+from dss.storage.hcablobstore import compose_blob_key
+from dss.storage.identifiers import DSS_BUNDLE_KEY_REGEX, VERSION_REGEX, UUID_REGEX, FILE_PREFIX
+from dss.storage.checkout.bundle import verify_checkout as bundle_checkout
+from dss.api.files import _verify_checkout as file_checkout
+from dss.storage.checkout import cache_flow
+
+logger = logging.getLogger(__name__)
+
+
+class CheckoutHandler:
+    def __init__(self, argv: typing.List[str], args: argparse.Namespace):
+        if args.keys is not None:
+            self.keys = args.keys.copy()
+        else:
+            self.keys = None
+        self.replica = Replica[args.replica]
+        self.handle = Config.get_blobstore_handle(self.replica)
+        self.bucket = self.replica.checkout_bucket
+
+    def process_command_locally(self, argv: typing.List[str], args: argparse.Namespace):
+        if self.keys is not None:
+            self.process_key(self.keys)
+
+    def _verify_delete(self, handle, bucket, key):
+        logger.warning(f'attempting removal of key: {bucket}/{key}')
+        try:
+            handle.delete(bucket=bucket, key=key)
+            handle.get(bucket=bucket, key=key)
+        except BlobNotFoundError:
+            logger.warning(f'Successfully deleted key {bucket}/{key}')
+        else:
+            raise RuntimeError(f'attempted to delete {bucket}/{key} but it did not work ;(')
+
+    def _verify_get(self, handle, bucket, key):
+        """ Helper Function to get file_metadata from cloud-blobstore"""
+        try:
+            file_metadata = json.loads(handle.get(bucket=bucket, key=key).decode('utf-8'))
+        except BlobNotFoundError:
+            logger.warning(f'unable to locate {bucket}/{key}')
+            return None
+        return file_metadata
+
+    def _verify_blob_existance(self, handle, bucket, key):
+        """Helper Function to see if blob data exists"""
+        try:
+            handle.get(bucket=bucket, key=key)
+        except BlobNotFoundError:
+            return False
+        return True
+
+    def _sleepy_checkout(self, function=None, **kwargs):
+        status = False
+        token = None
+        key = kwargs["file_metadata"]['name'] if kwargs.get('file_metadata') else kwargs.get("bundle_uuid")
+        logger.warning(f'starting {function.__name__} on {key}')
+        while status is not True:
+            kwargs['token'] = token
+            token, status = function(**kwargs)
+            if not status:
+                time.sleep(4)  # wait 10 seconds for checkout to complete, then try again
+
+    def _parse_key(self, key):
+        try:
+            version = VERSION_REGEX.search(key).group(0)
+            uuid = UUID_REGEX.search(key).group(0)
+        except IndexError:
+            RuntimeError(f'unable to parse the key {key}')
+        return uuid, version
+
+    def process_key(self, key):
+        raise NotImplementedError()
```

Comment:
If this is not implemented, does that mean any command with the `--keys` flag will not work? This raises an exception when running the `verify` action, not sure if it will break the `remove` or `start` actions (which I don't want to test out).

Reply:
@chmreid i saws some issues with the process_keys functionality, i've addressed them; and tested them manually, the integration_tests should also be passing